### PR TITLE
#245: Proxy Support

### PR DESCRIPTION
**Pull request by mvomiero**
Enhances: #245 
with recognition of **HTTP** and **HTTPS** proxies from `env variables` for `java http client`. Git requests recognize automatically the proxy env variables. 

* added package `com.devonfw.tools.ide.network` with class `ProxyContext` to store configurations about **http** and **https** proxies in two `ProxyConfig` objects for better encapsulation and code modularity.

### CODE REVIEW

**Comment by mvomiero** [comment]
* tested with fiddler (proxy address: `http://127.0.0.1:8888`) with native image, works properly and requests are detected

installhelm

fiddler

**Review by diiinesh — APPROVED** [review]
LGTM

File: cli/src/main/java/com/devonfw/tools/ide/network/ProxyContext.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      URL parsedUrl = new URL(url);
+      String protocol = parsedUrl.getProtocol();
+      if ("http".equalsIgnoreCase(protocol)) {
+        return httpProxyConfig;
+      } else if ("https".equalsIgnoreCase(protocol)) {
+        return httpsProxyConfig;
+      }
+    } catch (MalformedURLException e) {
+      this.context.warning(ProxyContext.PROXY_FORMAT_WARNING_MESSAGE);
+    }
```
[the remark this answers is not in the document]
Comment (mvomiero) [reply cli/src/main/java/com/devonfw/tools/ide/network/ProxyContext.java]: implemented!

**Comment by mvomiero** [comment]
Small Note on the tests: 
```
private static final String HTTP_PROXY = "http://127.0.0.1:8888";

assertThat("http:/" + proxy.address().toString()).isEqualTo(HTTP_PROXY);
```

The `toString()` method of `InetSocketAddress` (return value of `proxy.address()`) includes a leading slash to indicate that it is an IP address. This is a part of its toString() method implementation. That's why in the assert block is prefixed with `"http:/"` instead of `"http://"`

**Pull request merged**